GVFS/GVFS.Common/FileBasedDictionary.cs

```diff
@@ -9,16 +9,27 @@ namespace GVFS.Common
 {
     public class FileBasedDictionary<TKey, TValue> : FileBasedCollection
     {
-        private ConcurrentDictionary<TKey, TValue> data = new ConcurrentDictionary<TKey, TValue>();
+        private ConcurrentDictionary<TKey, TValue> data;
 
-        private FileBasedDictionary(ITracer tracer, PhysicalFileSystem fileSystem, string dataFilePath) 
+        private FileBasedDictionary(
+            ITracer tracer,
+            PhysicalFileSystem fileSystem, 
+            string dataFilePath, 
+            IEqualityComparer<TKey> keyComparer = null) 

```

Comment:
This is a private constructor, so do we actually need a default value for this parameter?

Reply:
Right. we don't need it. Updated.